eth/stagedsync/stage_hashstate.go

```diff
@@ -155,6 +155,52 @@ func unwindHashStateStageImpl(logPrefix string, u *UnwindState, s *StageState, t
 	return nil
 }
 
+// HashStateFromTo hashes the state for the PMT. Used in the zk sequencer node.
+// It is like SpawnHashStateStage, but it does not get the fromBlock from the executionAt because this happens mid-execution on zk sequencer.
+func HashStateFromTo(s *StageState, tx kv.RwTx, cfg HashStateCfg, from, to uint64, ctx context.Context, logger log.Logger) error {
```

Comment:
There is still quite a bit of code duplication here.  Could we call this `HashStateFromTo` function from the `SpawnHashStateStage` function to reduce the duplication?  It would also mean that the `HashStateFromTo` wouldn't need to manage the `useExternalTx` and instead just take a `tx` as a parameter.